File: javaGold/src/gold03/Sort.java
package gold03;

public class Sort {
    public static void main(String[] args) {
        int[] arr = {3, 1, 4, 5, 2};
        sort(arr);
        for (int i : arr) {
            System.out.println(i + " ");
        }
    }
    private static void sort(int[] arr) {
        for (int i = 0; i < arr.length - 1; i++) {
            for (int j = 0; j < arr.length - 1 - i; j++) {
                if (arr[j] > arr[j + 1]) {
                    int tmp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = tmp;
                }
            }
        }
    }
}

File: clang/lesson01/sort.c
#include <stdio.h>
#include <stdbool.h>

int main() {
    // 配列の作成
    int numbers[10];
    for (int i = 0; i < 10; i++) {
        int inputNum;
        printf("数値を入力してください\n");
        scanf("%d", &inputNum);
        numbers[i] = inputNum;
    }

    // sortのアルゴリズム
    for (int i = 0; i < 10; i++) {
        int maxIndex = 0;
        for (int k = 1; k < 10 - i; k++) {
            if (numbers[k] > numbers[maxIndex]) {
                maxIndex = k;
            }
        }
        int tmp = numbers[maxIndex];
        numbers[maxIndex] = numbers[9 - i];
        numbers[9 - i] = tmp;
    }

    // 配列の表示
    for (int i = 0; i < 10; i++) {
        printf("%d ", numbers[i]);
    }
}